### PR TITLE
USERINFO: include hwid in user's client string

### PR DESCRIPTION
**Pull request by ciscon**

### CODE REVIEW

File: sys_posix.c [hunk 1]
```diff
@@ -746,3 +746,22 @@ const char* Sys_HomeDirectory(void)
...
+                buffer[fread (buffer, 1, sizeof(buffer) - 1, f)] = '\0';
+                fclose (f);
+        } else if (f = fopen("/etc/hostid", "r")) {
+                buffer[fread (buffer, 1, sizeof(buffer) - 1, f)] = '\0';
+                fclose (f);
+        }
+	memcpy(hwid, buffer, sizeof(buffer));
+
+	return hwid;
+}
```
Comment (Calinou) [inline sys_posix.c:767]: This snippet has mixed tab and space indentation:

image

File: sys_posix.c [hunk 2]
```diff
@@ -746,3 +746,22 @@ const char* Sys_HomeDirectory(void)
 	}
 	return "";
 }
+
+char* Sys_HardwareID(void)
+{
+        FILE *f;
+        char buffer[255] = {0};
+	buffer[0] = '0';
+	char *hwid = malloc(sizeof(char)*255);
```
Comment (jite) [inline sys_posix.c:755]: sizeof(char) is always 1 and can be removed

File: sys_posix.c [hunk 3]
```diff
@@ -746,3 +746,22 @@ const char* Sys_HomeDirectory(void)
...
+
+        if (f = fopen("/etc/machine-id", "r")) {
+                buffer[fread (buffer, 1, sizeof(buffer) - 1, f)] = '\0';
```
Comment (jite) [inline sys_posix.c:758]: No need to terminate buffer as it's already zero-initialized

File: sys_posix.c [hunk 4]
```diff
@@ -746,3 +746,22 @@ const char* Sys_HomeDirectory(void)
...
+                fclose (f);
+        } else if (f = fopen("/etc/hostid", "r")) {
+                buffer[fread (buffer, 1, sizeof(buffer) - 1, f)] = '\0';
```
Comment (jite) [inline sys_posix.c:761]: same as above

File: sys_posix.c [hunk 5]
```diff
@@ -746,3 +746,22 @@ const char* Sys_HomeDirectory(void)
 	}
 	return "";
 }
+
+char* Sys_HardwareID(void)
+{
+        FILE *f;
+        char buffer[255] = {0};
```
Comment (jite) [inline sys_posix.c:753]: I don't see why you'd need the local buffer at all?

File: sys_posix.c [hunk 6]
```diff
@@ -746,3 +746,22 @@ const char* Sys_HomeDirectory(void)
...
+	buffer[0] = '0';
```
Comment (jite) [inline sys_posix.c:754]: You could do buffer[255] = {'0'}; above and the first entry would be ascii zero while the rest would be zero

File: cl_main.c [hunk 7]
```diff
@@ -1788,7 +1788,9 @@ static void CL_InitLocal (void)
 	Cvar_ForceSet (&cl_cmdline, com_args_original);
 	Cvar_ResetCurrentGroup();
 
-	snprintf(st, sizeof(st), "ezQuake %i", REVISION);
+	char *hwid = Sys_HardwareID();
```
Comment (jite) [inline cl_main.c:1791]: Variable declaration should be at top of scope

File: cl_main.c [hunk 8]
```diff
@@ -1788,7 +1788,9 @@ static void CL_InitLocal (void)
...
+
+	snprintf(st, sizeof(st), "ezQuake %i %s", REVISION, hwid);
```
Comment (jite) [inline cl_main.c:1793]: free hwid if not used anymore (goes out of scope when this function returns)

File: sys_posix.c [hunk 9]
```diff
@@ -746,3 +746,22 @@ const char* Sys_HomeDirectory(void)
...
+	char *hwid = malloc(sizeof(char)*255);
+
+        if (f = fopen("/etc/machine-id", "r")) {
+                buffer[fread (buffer, 1, sizeof(buffer) - 1, f)] = '\0';
+                fclose (f);
+        } else if (f = fopen("/etc/hostid", "r")) {
+                buffer[fread (buffer, 1, sizeof(buffer) - 1, f)] = '\0';
+                fclose (f);
+        }
+	memcpy(hwid, buffer, sizeof(buffer));
```
Comment (jite) [inline sys_posix.c:764]: can be removed if local buffer is removed

**Pull request closed without merging**